.taskcluster.yml

```diff
@@ -41,6 +41,16 @@ tasks:
           command: >-
             pip install pycodestyle &&
             pycodestyle ./mentoring
+        - name: pip-tools
+          image: python:3.9
+          command: >-
+            pip install pip-tools &&
+            ./pipupdate.sh &&
+            if ! output=$(git status --porcelain) || [ -n "$output" ]; then
+                echo "*** ./pipupdate.sh produced changes to the repository; these changes should be checked in ***";
+                git --no-pager diff;
+                exit 1;
+            fi
```

Comment:
I'm curious how this will work in practice. This means updates in your requirements of requirements (like ``certifi``) will cause tests to break. Will that be useful or annoying? I think an in-development project is a good place to find out.

Reply:
Because requirements.txt includes those transitive dependencies, they'll actually be pinned until we update them, and won't cause unexpected breakage.  But that reminds me, I need to enable some bot to do those updates.

Reply:
(this is a weird feature of `pip-update` btw -- it first reads from `requirements.txt` and then writes to it!)

Reply:
Thank you, that makes this check make a lot more sense. I removed it when converting [mozilla/ichnaea](https://github.com/mozilla/ichnaea) to ``pip-compile``, I'll need to look into adding it back.